src/gap.c

```diff
@@ -1523,6 +1523,22 @@ Obj FuncLOAD_DYN (
     if ( info == 0 )
         ErrorQuit( "call to init function failed", 0L, 0L );
 
+    // info->type should not be larger than kernel version
+    if (info->type / 10 > GAP_KERNEL_API_VERSION)
+        ErrorMayQuit("LOAD_DYN: kernel module built for newer "
+                     "version of GAP",
+                     0L, 0L);
+
+    // info->type should not have an older major version
+    if (info->type / 10000 < GAP_KERNEL_API_VERSION / 1000)
+        ErrorMayQuit("LOAD_DYN: kernel module built for older "
+                     "version of GAP",
+                     0L, 0L);
+
+    // info->type % 10000 should be 0, 1 or 2, for the 3 types of module
```

Comment:
shouldn't `10000` here be `10`?

Reply:
yes, that's why I shouldn't duplicate code in comments!